content/yaml-code-signing/code-signing-identities.md

```diff
@@ -0,0 +1,161 @@
+---
+title: Code signing identities
+description: How to set up code signing identities
+weight: 4
+aliases: /code-signing-yaml/code-signing-identities
+---
+
+Teams can use code signing identities to manage their code signing files in **Team settings > codemagic.yaml settings > Code signing identities** and reference them from their `codemagic.yaml` configuration. This makes managing code signing files easier and reduces the number of extra scripts required in your configuration file.
+
+## Managing and uploading files
+
+Team owner permissions are required to upload and edit files under the **Code signing identities** section. However, all team members can view the file info for any of the uploaded files.
+
+### iOS certificates
+
+Codemagic lets you upload code signing certificates as PKCS#12 archives containing both the certificate and private key which is needed to use it. When uploading, Codemagic will ask you to provide the certificate password (only enforced for password-protected certificates) along with a unique **Reference name**, which can then be used in the `codemagic.yaml` configuration to fetch the specific file.
+
+In addition, Codemagic provides the option to generate a new `Apple Development` or `Apple Distribution` certificate. Click on **Generate certificate**, provide a **Reference name** and choose the certificate type and a API key. Once the certificate has been created, Codemagic will allow you to download the certificate and provides the password for it. Note that the certificate can be downloaded just once right after creating it.
+
+Note that Apple limits the number of `Apple Distribution` certificates to 3. If you have already reached the maximum number of certificates, the following error will be displayed:
+
+```
+There is a problem with the request entity - You already have a current Distribution certificate or a pending certificate request.
+```
+
+Similar errors can also arise in rarer cases with `Apple Development` certificates. To resolve the error, either remove some old certificate from the Apple Developer Portal or upload an existing certificate manually.
+
+However, in the case of existing certificates generated by Codemagic, it is possible to automatically fetch them from the Apple Developer Portal based on the API key.
+
+{{<notebox>}}
+Fetching certificates that Codemagic has not created is disabled because each certificate has a unique private key to which Codemagic has no access, thus rendering the certificate unusable.
+{{</notebox>}}
+
+The **Reference name**, certificate type, team, and expiration date are displayed for each added certificate.
+
```

Comment:
We should mention that certificate generation and fetching are available only when the Apple Developer Portal integration is connected.